open-api/rest-catalog-open-api.yaml

```diff
@@ -3265,6 +3265,136 @@ components:
+          MUST NOT be applied beyond the entity identified by the Authentication header
+          (or other applicable authentication mechanism).
+      properties:
+        required-projections:
+          description: >
+            A list of projections that MUST be applied prior to any query-specified
+            projections.
+            If the required-projection property is absent, no mandatory projection applies,
+            and a reader MAY project any subset of columns of the table, including all columns.
+
+            1. A reader MUST project only columns listed in the required-projection.
+              - If a listed column has a transform, the reader MUST apply it and replace
+                all references to the underlying column with the transformed value
+                (for example, truncate[4](cc) MUST be projected as truncate[4](cc) AS cc,
+                and all references to cc during query evaluation post applying required-row-filter MUST resolve to this alias).
+              - Columns not listed in the required-projection MUST NOT be read.
+
+            2. A column MUST appear at most once in the required-projection.
+      
+            3. Multiple transformed versions of the same column (e.g., truncate[5](col)
+              and truncate[3](col) MUST NOT appear in the required-projection.
+      
+            4. If a projection entry includes an action that the reader cannot evaluate,
+              the reader MUST fail rather than ignore the transform.
+      
+            5. An identity transform is equivalent to projecting the column directly.
+            
+            8. The data type of the projected column MUST match the data type defined for the transform result.
+
+          type: array
+          items:
+            $ref: '#/components/schemas/Projection'
+        required-row-filter:
+          description: >
+            An expression that filters rows in the table.
+      
+            1. A reader MUST discard any row for which the filter evaluates to false or null, and
+              no information derived from discarded rows MAY be included in the query result.
+            
+            2. Row filters MUST be evaluated against the original, untransformed column values.
+              Required projections MUST be applied only after row filters are applied.
+      
+            3. If the catalog supports multiple row access filters for the table, it is
+              the catalog's responsibility to combine them using the appropriate logic (e.g., AND, OR).
+      
+            4. If a client cannot interpret or evaluate a provided filter expression, it MUST fail.
+      
+            5. If the required-row-filter property is absent or empty, no mandatory filtering is imposed.
+          $ref: '#/components/schemas/Expression'
+
+    Projection:
+      type: object
+      description: Defines a projection for a column.
+      properties:
+        source-id:
```

Comment:
I guess `source-id` is a reference to `PartitionField` but `source-id` is actually the list of arguments to apply to the transform in order to get the partition value? Shouldn't we use `field-id` instead?